gradle/buildSrc/src/main/kotlin/io/spine/gradle/internal/deps.kt

```diff
@@ -38,10 +40,48 @@ import java.net.URI
  *  https://github.com/uber/NullAway/blob/master/gradle/dependencies.gradle
  */
 
+/**
+ * A Maven repository.
+ */
 data class Repository(
     val releases: String,
     val snapshots: String,
-    val credentials: String
+    val credentialsFile: String? = null,
+    val credentials: Credentials? = null
+) {
+
+    fun credentials(project: Project): Credentials {
```

Comment:
Could you please comment on the expected form in which the credentials are expected to be laid out in a credentials file? I would see that as a part of the method-level documentation here.